File: Backend/Financeiro.ConciliacaoWebAPI/JsonWebToken/Code/Periodo/PeriodoBLL.cs
using JsonWebToken.Code.Periodo;
using JsonWebToken.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace JsonWebToken.Code
{
  public class PeriodoBLL
  {
    public PeriodoModel selectPeriodoPorId(int id) {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
        return DAL.selectPeriodoPorId(id);
      }
      catch (Exception e)
      {
        throw e;
      }
    }
    public List<PeriodoModel> selectPeriodo(string nome,string situacao)
    {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
        List<PeriodoModel> list = DAL.selectPeriodo(nome,situacao);
        return list;
      }
      catch (Exception e)
      {
        throw e;
      }
    }
  

    public void deletePeriodo(int id)
    {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
      DAL.deletePeriodo(id);

      }
      catch (Exception e)
      {
        throw e;
      }
    }
    public void insertPeriodo(PeriodoModel periodo)
    {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
        DAL.inserirPeriodo(periodo);

      }
      catch (Exception e)
      {
        throw e;
      }

    }

    public void updatePeriodo(PeriodoModel periodo)
    {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
        DAL.updatePeriodo(periodo);
      }
      catch (Exception e)
      {
        throw e;
      }

    }

    public void ativarDesativarPeriodo(PeriodoModel periodo)
    {
      try
      {
        PeriodoDAL DAL = new PeriodoDAL();
        if (periodo.situacao.Equals("0"))
        {
          DAL.ativarPeriodo(periodo.codPeriodo);
        }
        else {
          DAL.desativarPeriodo(periodo.codPeriodo);
        }
      }
      catch (Exception e)
      {
        throw e;
      }

    }
  }
}

File: Backend/Administracao.ConciliacaoAPI/Code/Periodo/PeriodoBLL.vb
Imports Administracao.ConciliacaoAPI.Model

Namespace Code.Periodo
    Public Class PeriodoBLL
        Public Function selectPeriodoPorId(id As Integer) As PeriodoModel
            Try
                Dim DAL As New PeriodoDAL()
                Return DAL.selectPeriodoPorId(id)
            Catch e As Exception
                Throw e
            End Try
        End Function
        Public Function selectPeriodo(nome As String, situacao As String) As List(Of PeriodoModel)
            Try
                Dim DAL As New PeriodoDAL()
                Dim list As List(Of PeriodoModel) = DAL.selectPeriodo(nome, situacao)
                Return list
            Catch e As Exception
                Throw e
            End Try
        End Function


        Public Sub deletePeriodo(id As Integer)
            Try
                Dim DAL As New PeriodoDAL()

                DAL.deletePeriodo(id)
            Catch e As Exception
                Throw e
            End Try
        End Sub
        Public Sub insertPeriodo(periodo As PeriodoModel)
            Try
                Dim DAL As New PeriodoDAL()

                DAL.inserirPeriodo(periodo)
            Catch e As Exception
                Throw e
            End Try

        End Sub

        Public Sub updatePeriodo(periodo As PeriodoModel)
            Try
                Dim DAL As New PeriodoDAL()
                DAL.updatePeriodo(periodo)
            Catch e As Exception
                Throw e
            End Try

        End Sub

        Public Sub ativarDesativarPeriodo(periodo As PeriodoModel)
            Try
                Dim DAL As New PeriodoDAL()
                If periodo.situacao.Equals(0) Then
                    DAL.ativarPeriodo(periodo.codPeriodo)
                Else
                    DAL.desativarPeriodo(periodo.codPeriodo)
                End If
            Catch e As Exception
                Throw e
            End Try

        End Sub
    End Class
End Namespace
